src/tests/unit/endpoint_standard/test_usb_device_approval.py

```diff
@@ -0,0 +1,189 @@
+    assert approval.vendor_id == "0x0781"
+    assert approval.vendor_name == "SanDisk"
+    assert approval.product_id == "0x5581"
+    assert approval.product_name == "Ultra"
+    assert approval.approval_name == "Example Approval"
+    approval.delete()
+
+
+def test_approval_query_with_all_bells_and_whistles(cbcsdk_mock):
+    """Tests running the query with all values set."""
+    def post_validate(url, body, **kwargs):
+        assert body['query'] == '*'
+        crits = body['criteria']
+        assert crits['device.id'] == ['A1000', 'B1000', 'C1000']
+        assert crits['vendor_name'] == ['SanDisk']
+        assert crits['product_name'] == ['Ultra']
+        return USBDEVICE_APPROVAL_QUERY_RESP
+
+    cbcsdk_mock.mock_request("POST", "/device_control/v3/orgs/test/approvals/_search", post_validate)
+    api = cbcsdk_mock.api
+    query = api.select(USBDeviceApproval).where('*').set_device_ids(['A1000', 'B1000']).set_vendor_names(['SanDisk']) \
+                                         .set_product_names(['Ultra']).set_device_ids(['C1000'])
+    assert query._count() == 1
+    results = [result for result in query._perform_query()]
+    assert len(results) == 1
+    approval = results[0]
+    assert approval._model_unique_id == "10373"
+    assert approval.vendor_id == "0x0781"
+    assert approval.vendor_name == "SanDisk"
+    assert approval.product_id == "0x5581"
+    assert approval.product_name == "Ultra"
+    assert approval.approval_name == "Example Approval"
+
+
+def test_approval_query_async(cbcsdk_mock):
+    """Test running the query in asynchronous mode."""
+    cbcsdk_mock.mock_request("POST", "/device_control/v3/orgs/test/approvals/_search", USBDEVICE_APPROVAL_QUERY_RESP)
+    api = cbcsdk_mock.api
+    future = api.select(USBDeviceApproval).where('*').execute_async()
+    results = future.result()
+    assert len(results) == 1
+    approval = results[0]
+    assert approval._model_unique_id == "10373"
+    assert approval.vendor_id == "0x0781"
+    assert approval.vendor_name == "SanDisk"
+    assert approval.product_id == "0x5581"
+    assert approval.product_name == "Ultra"
+    assert approval.approval_name == "Example Approval"
+
+
+def test_approval_query_parameters_validate_fail(cbcsdk_mock):
+    """Test that we get the appropriate exceptions when we pass in bogus parameters."""
+    query = cbcsdk_mock.api.select(USBDeviceApproval)
+    with pytest.raises(ApiError):
+        query.set_device_ids([1])
+    with pytest.raises(ApiError):
+        query.set_vendor_names([2])
+    with pytest.raises(ApiError):
+        query.set_product_names([3])
+
+
+def test_approval_bulk_create(cbcsdk_mock):
+    """Test the bulk_create function."""
+    cbcsdk_mock.mock_request("POST", "/device_control/v3/orgs/test/approvals/_bulk",
+                             USBDEVICE_APPROVAL_BULK_CREATE_RESP)
+    api = cbcsdk_mock.api
+    results = USBDeviceApproval.bulk_create(api, USBDEVICE_APPROVAL_BULK_CREATE_REQ)
```

Comment:
This should be two items since this is a bulk request

Reply:
I don't have example data for creating two items; the example data in the developer documentation only shows one.

Reply:
You can make up another item it doesn't have to be 100% accurate

Reply:
OK